internal/rbd/rbd_util.go

```diff
@@ -1934,13 +1945,22 @@ func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
 	return nil
 }
 
-// setSnapshotMetadata Set snapshot-name/snapshot-namespace/snapshotcontent-name metadata
-// on RBD image.
-func (rv *rbdVolume) setSnapshotMetadata(parameters map[string]string) error {
-	for k, v := range k8s.GetSnapshotMetadata(parameters) {
-		err := rv.SetMetadata(k, v)
-		if err != nil {
-			return fmt.Errorf("failed to set metadata key %q, value %q on image: %w", k, v, err)
+// unsetAllMetadata unset all the metadata from arg keys on RBD image.
+func (rv *rbdVolume) unsetAllMetadata(keys []string) error {
+	list, err := rv.ListMetadata()
+	if err != nil {
+		return fmt.Errorf("failed to list metadata image %s/%s: %w", rv.Pool, rv.RbdImageName, err)
+	}
+
+	for k, v := range list {
+		for _, key := range keys {
+			if k != key {
+				continue
+			}
+			err = rv.RemoveMetadata(key)
+			if err != nil && !strings.Contains(err.Error(), "No such file or directory") {
+				return fmt.Errorf("failed to unset metadata key %q, value %q on %q: %w", k, v, rv, err)
+			}
```

Comment:
To unset why we need to list and remove it? just do RemoveMetadata and ignore not/no found error?